taichi/backends/cc/cc_program.h

```diff
@@ -42,23 +68,30 @@ class CCProgram {
     return runtime.get();
   }
 
+  ~CCProgramImpl() {
+  }
+
+  CCFuncEntryType *load_kernel(std::string const &name);
+  void compile_layout(SNode *root);
+  void relink();
+
   CCContext *update_context(Context *ctx);
   void context_to_result_buffer();
 
-  Program *const program;
-
  private:
-  std::vector<char> args_buf;
-  std::vector<char> root_buf;
-  std::vector<char> gtmp_buf;
+  void add_kernel(std::unique_ptr<CCKernel> kernel);
+  void init_runtime();
+
   std::vector<std::unique_ptr<CCKernel>> kernels;
   std::unique_ptr<CCContext> context;
   std::unique_ptr<CCRuntime> runtime;
   std::unique_ptr<CCLayout> layout;
   std::unique_ptr<DynamicLoader> dll;
   std::string dll_path;
+  std::vector<char> args_buf;
```

Comment:
underscore suffix?

Reply:
I'll modify these in another PR, otherwise, it would be a LOT of files change.